runtimes/alibi-detect/mlserver_alibi_detect/cd/runtime.py

```diff
@@ -0,0 +1,42 @@
+from mlserver_alibi_detect.runtime import AlibiDetectRuntime
+from mlserver_alibi_detect.errors import InvalidAlibiDetector
+from mlserver.utils import get_model_uri
+from alibi_detect.utils.saving import load_detector
+from importlib import import_module
+from typing import Any
+import pickle
+
+ALIBI_DETECT_DRIFT_MODULE = "alibi_detect.cd"
+
+
+class AlibiDriftDetectRuntime(AlibiDetectRuntime):
+    """
+    Implementation of the MLModel interface to load and serve Alibi drift models.
+    """
+
+    async def load(self) -> bool:
+
+        model_uri = await get_model_uri(self._settings)
+
+        if not bool(self.alibi_detect_settings.init_detector):
+            self._model = load_detector(model_uri)
+        else:
+            detector_data = pickle.load(open(model_uri, "rb"))
+            try:
+                detector_type = self.alibi_detect_settings.detector_type
+
+                detect_module = import_module(ALIBI_DETECT_DRIFT_MODULE)
+                drift_detector = getattr(detect_module, detector_type)
+            except Exception as e:
+                raise InvalidAlibiDetector(detector_type, self._settings.name, e)
```

Comment:
Instead of passing the underlying exception manually, we could "chain" the exceptions `e` and `InvalidAlibiDetector()` as:

```
raise InvalidAlibiDetector(detector_type, self._settings.name) from e
```

That way we preserve all the information (e.g. stacktrace, etc.). First time I hear about this BTW... full credit goes to this [SO answer](https://stackoverflow.com/a/792163/5015573).

Reply:
Sure will update